### DISCUSSION TITLE
error:0909006C:PEM routines:get_name:no start line

### DISCUSSION THREAD

**Discussion by babitakapoor112** [openssl/openssl#23521] [Q&A]
I have created private and public keys at my local machine using following commands:

1. openssl genrsa -out private.pem 2048

2. openssl rsa -in private.pem -pubout -outform PEM -out public_key.pem

3. openssl pkcs8 -topk8 -inform PEM -in private.pem -out private_key.pem -nocrypt
and used these keys to sign and verify my signature. Currently I am facing the issue which is invalid signature. When I used openssl_error_string() to print the error in PHP it is gving me error:0909006C:PEM routines:get_name:no start line in return. 
Moreover, I checked my public and private key initials, which are -----BEGIN RSA PRIVATE KEY---- and -----END RSA PRIVATE KEY-----
I am bit confused right now what is the main issue in verifying. Is it because of the local environment or something else?
Also, my signature header is t:1707392347,s:Bk1M8RnmE/NNXZK5TIayhf8cMIsxTD+focn7ISOsTJD30Qn0F8ktCBhFju4pjyPYEgPiS8h3BtIDeQ44u+21h2GOYj/JP32hdGQX4vMe/P88vRaH4r4lSwrelnVXXF7mR2LDg7ktIxKVpzIs472rP1aZqd2QC0WvY1Pd9nYQQYE24wWwHCuOCGZiBCtTfOp28rrE9a1vqsjjRg+vE4KfzEHG3tvz78A4rlTeGphSQXaHay+QNtu5SwRfRmztIDTZ3MaecWIQJCA4JSjFHMEnN8C4DKHdwYl8nGlIIHUh7uPb8xp4kb+9C7VaiIOR1C2TZK17mEHz08c7/YNoFUEv0Q==

Do I need to pass something extra in header for the request?

**Comment by nhorman** [comment]
The error is raised in get_name if the input BIO fails to read a line from the input file.  the underlying BIO_fgets maps to an fgets call, which would ostensibly be the source of the failure.  First thought would be that your web server running the php code is being run as a user that doesn't have permission to access the private key in question.

**Reply by babitakapoor112** [reply]
I have resolved the issue by the way. The issue was related to the type of payload I was passing for my signature. But now it is resolved.

**Comment by babitakapoor112** [comment]
So I am closing this discussion.
